src/pages/ZNS/ZNS.tsx

```diff
@@ -257,6 +257,15 @@ const ZNS: React.FC<ZNSProps> = ({ domain, version, isNftView: nftView }) => {
 		}
 		if (triedEagerConnect)
 			addNotification(active ? 'Wallet connected.' : 'Wallet disconnected.');
+
+		// Check if we need to close a modal
+		if (
+			(!active && modal === Modal.Profile) ||
+			modal === Modal.Transfer ||
```

Comment:
This logic will close the modal if the transfer or mint modal is open, regardless of `active`. Is this the intended behavior?

(!active && (modal === Modal.Profile || modal === Modal.Transfer || modal === Modal.Mint)) makes more sense here.